crates/holochain/src/core/ribosome/real_ribosome.rs

```diff
@@ -814,38 +900,12 @@ impl RibosomeT for RealRibosome {
 
         match zome.zome_def() {
             ZomeDef::Wasm(_) => {
-                let module = self.module(zome.zome_name())?;
-
-                // Check if the wasm has a function that matches this type.
-                if module.exports().functions().any(|f| {
-                    f.name() == name
-                        && f.ty().params().is_empty()
-                        && f.ty().results() == [Type::I32]
-                }) {
-                    let (instance, context_key) = self.instance(call_context)?;
-
-                    // Call the function as a native function.
-                    let result = instance
-                        .lock()
-                        .exports
-                        .get_native_function::<(), i32>(name)
-                        .ok()
-                        .map_or(Ok(None), |func| Ok(Some(func.call()?)))
-                        .map_err(|e: RuntimeError| {
-                            RibosomeError::WasmRuntimeError(
-                                wasm_error!(WasmErrorInner::Host(format!("Failed during call to wasm function. Zome Name: {}, Fn Name: {}, Error: {}", zome.zome_name(), name, e))).into(),
-                            )
-                        })?;
-
-                    // Remove the blank context.
-                    CONTEXT_MAP.lock().remove(&context_key);
-
-                    Ok(result)
-                } else {
-                    // the func doesn't exist
-                    // the callback is not implemented
-                    Ok(None)
-                }
+                let module = self.runtime_compiled_module(zome.zome_name())?;
+                self.get_const_fn_for_wasm(call_context, name, module)
+            }
+            ZomeDef::WasmDylib(zome_wasm_dylib) => {
+                let module = self.precompiled_module(&zome_wasm_dylib.path)?;
```

Comment:
@maackle the key difference is for the dylib setup to call precompiled_module instead of runtime_compiled_module